frontend/app/src/components/AppContext.tsx

```diff
@@ -79,6 +79,13 @@ export interface Props {
    */
   sidebarChevronDownshift: number
 
+  /**
+   * Whether to disable the sidebar page navigation links, based on connection
+   * state and whether the host has disabled inputs.
+   * @see SidebarNavLink
+   */
+  disableSidebarNavLinks: boolean
```

Comment:
nit: I think it might be a little bit more useful if we directly put the `widgetsDisabled` into the context  to be reusable across other components / use case. Currently, this state is only used by other components by passing this through as props. Also we currently only have a single case for `disableSidebarNavLinks` which is when all widgets should be disabled.